public/components/custom_panels/helpers/utils.tsx

```diff
@@ -136,13 +166,110 @@ export const onTimeChange = (
   setEnd: React.Dispatch<React.SetStateAction<string>>
 ) => {
   const recentlyUsedRange = recentlyUsedRanges.filter((recentlyUsedRange) => {
-    const isDuplicate = recentlyUsedRange.start === start && recentlyUsedRange.end === end;
+    const isDuplicate =
+      recentlyUsedRange.start === start && recentlyUsedRange.end === end;
     return !isDuplicate;
   });
   recentlyUsedRange.unshift({ start, end });
   setStart(start);
   setEnd(end);
   setRecentlyUsedRanges(
-    recentlyUsedRange.length > 10 ? recentlyUsedRange.slice(0, 9) : recentlyUsedRange
+    recentlyUsedRange.length > 10
+      ? recentlyUsedRange.slice(0, 9)
```

Comment:
i don't think the length check is necessary? `.slice` should check it

Reply:
makes sense! will change this.